qcodes/tests/dataset/test_measurement_context_manager.py

```diff
@@ -245,6 +245,24 @@ def test_log_includes_extra_info(caplog, meas_with_registered_param):
 
     assert "some extra info" in caplog.text
 
+def test_nested_measurement_throws_error(experiment, DAC, DMM):
+    meas1 = Measurement()
+    meas2 = Measurement()
+
+    # pytest.raises(Exception): does not work because it does not allow
+    # the state of _is_entered to be changed to False when context manager
+    # ends. Hence all the test after this one fails.
+    try:
+        with meas1.run():
+            with meas2.run():
+                pass
+            pass
+    except RuntimeError:
+        return True
```

Comment:
in order to mimic the behavior of `pytest.raises` we'd need more, smth like this:
```suggestion
    exception_raised = False

    try:
        with meas1.run():
            with meas2.run():
                pass
            pass
    except RuntimeError as e:
        if str(e) == 'Nested measurements are not supported':
            exception_raised = True
        else:
            raise e

    assert exception_raised, "Expected exception about nesting measurements wasn't raised"
```

Reply:
Tests are still failing. So pytest.raises(Excpetion) is not the only reason, although not using it fixes the few tests. I do not completely understand why things are failing so I will look into this again afterwards.